modules/cas_cache/utils/utils_mpool.c

```diff
@@ -98,20 +124,24 @@ void *cas_mpool_new_f(struct cas_mpool *mpool, uint32_t count, int flags)
 	return items;
 }
 
-void *cas_mpool_new(struct cas_mpool *mpool, uint32_t count)
+void *env_mpool_new(struct env_mpool *mpool, uint32_t count)
 {
-	return cas_mpool_new_f(mpool, count, mpool->flags);
+	return env_mpool_new_f(mpool, count, mpool->flags);
 }
 
-void cas_mpool_del(struct cas_mpool *mpool,
+bool env_mpool_del(struct env_mpool *mpool,
 		void *items, uint32_t count)
 {
 	env_allocator *allocator;
 
-	allocator = cas_mpool_get_allocator(mpool, count);
+	allocator = env_mpool_get_allocator(mpool, count);
 
 	if (allocator)
 		env_allocator_del(allocator, items);
-	else
+	else if (mpool->fallback)
 		cas_vfree(items);
+	else
+		return false;
```

Comment:
ENV_BUG()? It seems that nobody checks the return value anyways, so you can change it back to void

Reply:
OCF will use it while freeing requests, so it can determine whether it also has to free the map, but maybe I'll handle all the ocf_request-specific code in ocf_request instead of making mpools overly complicated.

Reply:
I left it after all, let me know what you think about it.

Reply:
ok